src/CloudNative.CloudEvents/CloudEventContent.cs

```diff
@@ -32,6 +34,8 @@ public CloudEventContent(CloudEvent cloudEvent, ContentMode contentMode, ICloudE
             if (contentMode == ContentMode.Structured)
             {
                 inner = new InnerByteArrayContent(formatter.EncodeStructuredEvent(cloudEvent, out var contentType));
+                // This is optional in the specification, but can be useful.
+                MapHeaders(cloudEvent, includeDataContentType: true);
                 Headers.ContentType = new MediaTypeHeaderValue(contentType.MediaType);
```

Comment:
Not related to these changes, but rather than checking
```
if (cloudEvent.Data is byte[])
```
should we check for IEnumerable<byte>? Also what about ReadOnlyMemory<byte>?

Reply:
Basically that all comes back to the question of how we handle data. I'd like to get this (and some other minor fixes) in before we tackle that, if possible.